src/main/java/org/itsallcode/io/CapturingOutputStream.java

```diff
@@ -0,0 +1,131 @@
+package org.itsallcode.io;
+
+/*-
+ * #%L
+ * JUnit5 System Extensions
+ * %%
+ * Copyright (C) 2016 - 2018 itsallcode.org
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * This class implements an output stream that can capture the data put into it
+ * on demand. It is intended for testing purposes where the tester needs to
+ * snoop into an output streams content.
+ *
+ * <p>
+ * When provided with a target output stream, it additionally forwards all data
+ * to that stream, working like a "tee" for the streams. The "tee" feature
+ * allows the program under test to operate normally.
+ * </p>
+ *
+ * <p>
+ * To avoid capturing unwanted data, capturing must be explicitly enabled by
+ * calling {@link #capture()}.
+ */
+public class CapturingOutputStream extends OutputStream
+{
+    OutputStream targetStream;
+    ByteArrayOutputStream internalStream;
+    private boolean capture;
+
+    public CapturingOutputStream(final OutputStream targetStream)
+    {
+        this.targetStream = targetStream;
+        resetCapturing();
+    }
+
+    @Override
+    public void write(final int b) throws IOException
+    {
+        if (this.capture)
+        {
+            this.internalStream.write(b);
```

Comment:
Should we also write to target stream when capturing?

Reply:
Imho: yes, otherwise the behavior of the system under test changes.